### ISSUE TITLE
Generators are only available when targeting ECMAScript6 or higher

### ISSUE THREAD

**Issue by kosak** [Microsoft/nodejstools#947]
Hi, hopefully this is an easy question for someone. It has been driving me crazy.

I am unable to get IntelliSense for ES6 features to work in Visual Studio 2015 Update 2. I have seen several other hits on the Internet for the same problem and as far as I can tell, people seem to be able to resolve it. Somehow I personally continue to fail. I've tried this a few times on multiple machines but I'm probably doing the same thing wrong every time.

Any idea what I'm doing wrong here? There's probably some little box I'm not checking somewhere.

**Thanks in advance for your help!**

Repro:

This is a fresh computer with fresh installs of everything: Windows 10, Visual Studio 2015 Update 2, NodeJS Tools 1.2 Alpha 2, TypeScript 1.8 for Visual Studio 2015 downloaded from typescriptlang.org, nodejs v0.12.14 downloaded from nodejs.org

Inside Visual Studio I do

New project (Other Languages -> TypeScript -> Node.js -> Blank Node.js Console Application)

In Solution Explorer, right click project, project properties -> TypeScript Build -> ECMAScriptVersion -> ECMAScript 6 [(link to screenshot)](http://imgur.com/BkVWBtX)

Tools -> Options -> Text Editor -> Node.js -> IntelliSense
Look at the radio button ES6 Intellisense Preview (already checked) [(link to screenshot)](http://imgur.com/rl3Qi4T)

In the existing app.ts file I type

```
function* foo() {
    yield 4;
}

function bar() {
    const x = 5;
    console.log(`Hello: ${x}`);
}
```

Close and reopen Visual Studio 2015 for good measure.

What I see: "Generators are only available when targeting ECMAScript6 or higher" [(link to screenshot):](http://imgur.com/KGx0Ptv)

**_Possibly important clue:**_ If I look at the .js file that is created when I press "save", the output is definitely not ES6. My evidence for this is the fact that `const` was turned into `var` and my string interpolation was done the old way. So is it possible there is some code generation setting I missed somewhere? The generated app.js file looks like this:

```
$ cat app.js
function foo() {
    yield 4;
}
function bar() {
    var x = 5;
    console.log("Hello: " + x);
}
//# sourceMappingURL=app.js.map
```

`function bar` is some previous version of JavaScript (not 6) and `function foo` as far as I can tell is just the compiler giving up and throwing away the \* in my `function*`.

These are my version numbers from Help -> About...

```
Microsoft Visual Studio Professional 2015
Version 14.0.25123.00 Update 2
Node.js Tools   1.2.40329.01
Node.js Tools - Profiling   1.2.40329.01
TypeScript   1.8.31.0
TypeScript tools for Visual Studio
```

Again, I'd really appreciate anyone's help. Thanks for reading this far!

**Comment by mjbvz** [comment]
Thanks for the detailed report. I can repo the problem as well and will keep looking into the root cause. It may be Typescript, or it may be how NTVS interfaces with Typescript.

In the meantime, try adding a `tsconfig.json` file to the top level of your project with the content:

```json
{
  "compilerOptions": { "target": "es6" }
}
```

This generates the correct JavaScript output on build, but does seem to disable the compile on save option (even after adding `"compileOnSave": true` to the `tsconfig.json`.)

**Comment by kosak** [comment]
Hiya, thanks for looking into this! I wonder if you have had any luck finding out why the ES6 IntelliSense isn't working. Randomly, I tried running Process Explorer to see if I could find any invocations of tsc.exe, but judging from the info on devenv.exe, it looks to me like it directly links to things like

```C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\CommonExtensions\Microsoft\TypeScript\Microsoft.VisualStudio.LanguageServices.TypeScript.dll```

I don't know what that is, I don't know how to peer inside it, and so I don't know if that's how it's invoking the compiler to do error analysis.

So basically I'm grasping at straws. :anguished:

**Comment by paulvanbrenk** [comment]
@kosak is this still an issue when you use VS 2015 Update 3, TypeScript 2.1, and the RTW version of Node Tools 1.2?

If so please re-open

**Comment by kosak** [comment]
Magnificent. Working great now. Thank you!!!

(sorry for my delay in responding)
